### PR TITLE
[2/2] Feature(favicon): Add Favicon resources from the backend to be added to the frontend

### PR DESCRIPTION
**Pull request by hanywang2**
# Summary

This PR adds favicon resource URIs generated from the backend (see https://github.com/mintlify/server/pull/9) to be appended onto the frontend. It also removes former dependencies of the `favicon` package on the frontend and the `injectFavicon` function on prebuild

### CODE REVIEW

File: client/src/pages/_sites/[site]/[[...slug]].tsx [hunk 1]
```diff
@@ -198,6 +254,7 @@ export const getStaticProps: GetStaticProps<PageProps, PathProps> = async ({ par
         config,
         openApi,
       }),
+      stringifiedFavicons: stringify(favicons),
```
Comment (hahnbeelee) [inline client/src/pages/_sites/[site]/[[...slug]].tsx:257]: if each url for a favicon is going to be following a certain format do we need to be sending favicon data?

Comment (hanywang2) [reply client/src/pages/_sites/[site]/[[...slug]].tsx:257]: I did it in case it'll change

File: client/src/pages/_sites/[site]/[[...slug]].tsx [hunk 2]
```diff
@@ -65,12 +67,26 @@ interface ParsedDataProps {
...
   openApi?: any;
 }
+
+interface FaviconsProps {
+  'apple-touch-icon': string;
+  '32x32': string;
+  '16x16': string;
+  'shortcut-icon': string;
+  browserconfig: string;
+}
```
Comment (hahnbeelee) [inline client/src/pages/_sites/[site]/[[...slug]].tsx:77]: Did you cherry pick these particular types of files? A better thing to do might be to make `FaviconProps` type `{ rel: string, href: string, sizes: string }[]` and inside the `<Head>`:
```js
favicons.map((fav) => {
  return (
    <link rel={fav.rel} sizes={fav.sizes} href={fav.href} />
  )
}
```

File: client/src/utils/api.ts [hunk 3]
```diff
@@ -58,16 +58,16 @@ const potentiallAddPathParams = (inputUrl: string, inputData: Record<string, any
...
-  if (contentType === 'multipart/form-data') {
-    let cleanedObj = removeEmpty(obj);
-    const bodyFormData = new FormData();
-    for (var key in cleanedObj) {
-      bodyFormData.append(key, cleanedObj[key]);
-    }
-
-    return bodyFormData;
-  }
+const getBody = (obj: Object, _: string) => {
```
Comment (hahnbeelee) [inline client/src/utils/api.ts:61]: what is this

**Review by hahnbeelee — CHANGES_REQUESTED** [review]
Left some comments on both the front and backend. I think you either need to include all the different file types being generated on the frontend or stop generating them on the backend

File: client/src/utils/api.ts (hunk 3 above)
Comment (hanywang2) [reply client/src/utils/api.ts:61]: A rebase from main

File: client/src/pages/_sites/[site]/[[...slug]].tsx (hunk 2 above)
Comment (hanywang2) [reply client/src/pages/_sites/[site]/[[...slug]].tsx:77]: That's a good point

**Review by hahnbeelee — APPROVED** [review]

**Pull request merged**